src/benchmarking.rs

```diff
@@ -330,57 +273,98 @@ benchmarks! {
 
-		Tellor::<T>::submit_value(
-			RawOrigin::Signed(reporter.clone()).into(),
-			staking_to_local_token_query_id,
-			uint_value::<T>(6 * 10u128.pow(18)),
-			0,
-			staking_to_local_token_query_data)?;
+		let reports = 2u32.saturating_pow(s);
+		for i in 1..=reports {
+			Tellor::<T>::submit_value(
+				RawOrigin::Signed(reporter.clone()).into(),
+				staking_token_price_query_id,
+				uint_value::<T>(50 * 10u128.pow(18)),
+				0,
+				staking_token_price_query_data.clone())?;
+			if i > 1 && i < reports {
+				Tellor::<T>::begin_dispute(RawOrigin::Signed(disputer.clone()).into(),
+					staking_token_price_query_id,
+					<ReportedTimestampsByIndex<T>>::get(staking_token_price_query_id, i - 1).unwrap(),
+					Some(address))?;
+			}
+			T::BenchmarkHelper::set_time(12 * HOURS);
+		}
 
-		T::BenchmarkHelper::set_time(12 * HOURS);
+		let reports = 2u32.saturating_pow(l);
+		for i in 1..=reports {
+			Tellor::<T>::submit_value(
+				RawOrigin::Signed(reporter.clone()).into(),
+				staking_to_local_token_query_id,
+				uint_value::<T>(6 * 10u128.pow(18)),
+				0,
+				staking_to_local_token_query_data.clone())?;
+			if i > 1 && i < reports {
+				Tellor::<T>::begin_dispute(RawOrigin::Signed(disputer.clone()).into(),
+					staking_to_local_token_query_id,
+					<ReportedTimestampsByIndex<T>>::get(staking_to_local_token_query_id, i - 1).unwrap(),
+					Some(address))?;
+			}
+			T::BenchmarkHelper::set_time(12 * HOURS);
+		}
 	}: _(RawOrigin::Signed(reporter))
 
 	begin_dispute {
+		// Maximum number of sequential disputed timestamps
+		let d in 1..T::MaxDisputedTimeSeries::get();
 		let query_data: QueryDataOf<T> = BoundedVec::try_from(vec![0u8; T::MaxQueryDataLength::get() as usize]).unwrap();
 		let query_id = Keccak256::hash(query_data.as_ref()).into();
-		let reporter = account::<AccountIdOf<T>>("account", 1, SEED);
-		let disputer = account::<AccountIdOf<T>>("account", 2, SEED);
-		let address = Address::zero();
-		deposit_stake::<T>(reporter.clone(), trb(1_200), address)?;
-		T::BenchmarkHelper::set_balance(disputer.clone(), 1_000);
+		let stake_amount = <StakeAmount<T>>::get();
+		let dispute_fees = token::<T>(10u8);
 
-		T::BenchmarkHelper::set_time(HOURS);
-		Tellor::<T>::submit_value(RawOrigin::Signed(reporter).into(),
-			query_id,
-			uint_value::<T>(4_000),
-			0,
-			query_data
+		T::BenchmarkHelper::set_time(REPORTING_LOCK);
+
+		// Create initial timestamp for later dispute in extrinsic call
+		let reporter = account::<AccountIdOf<T>>("account", 0, SEED);
+		deposit_stake::<T>(reporter.clone(), stake_amount, Address::zero())?;
+		T::BenchmarkHelper::set_balance(reporter.clone(), dispute_fees);
+		Tellor::<T>::submit_value(
+			RawOrigin::Signed(reporter.clone()).into(), query_id, uint_value::<T>(10), 0, query_data.clone()
 		)?;
 
-		let report_timestamps = <TimeOfLastNewValue<T>>::get().unwrap();
+		// Create series of disputed timestamps, using new accounts to avoid reporting lock
+		for i in 2..d {
```

Comment:
can we make it like this `for i in 1..=d` as in the minimum range eg(d = 1) we'll not get any iterations?

Reply:
The first dispute is created in the code block above, so there is always at least one dispute.